DOCSP-34578 Vector search index (#598)
mongodb/docs-compass: source/indexes/create-vector-search-index.txt

.. _compass-create-vector-search:

===================================
Create an Atlas Vector Search Index
===================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

.. versionadded:: 1.42.1

You can create Atlas Vector Search indexes using |compass|. These indexes 

> enable you to index vector data *and* other data types, facilitating 

Reviewer: [suggestion] 
```suggestion
enable you to index vector data *and* other data types, which facilitates 
```

Author: Updated.

semantic searches on the indexed fields. Atlas Vector Search indexes support 

> indexing vector data by identifying the most similar vectors. This is achieved 
> by calculating the distance between the query vector and the vectors stored 
> in the index.

Reviewer: [suggestion to [avoid starting the sentence with "This"](https://www.mongodb.com/docs/meta/style-guide/writing/use-pronouns-carefully/#this)]

```suggestion
indexing vector data by identifying the most similar vectors. The index determines similarity 
by calculating the distance between the query vector and the vectors stored 
in the index.
```

About this Task
---------------

> To create an Atlas Search index, your deployment must be either:

Reviewer: [suggestion] to [avoid introducing a list with a fragment ](https://www.mongodb.com/docs/meta/style-guide/style/lists/#introductory-text)

```suggestion
To create an Atlas Search index, your deployment must align with one of the following cases:
```

Author: updated.

> - Hosted on `MongoDB Atlas <https://www.mongodb.com/docs/atlas/>`__ and
>   have an Atlas cluster tier of M10 or higher.

Reviewer: ```suggestion
- A deployment hosted on `MongoDB Atlas <https://www.mongodb.com/docs/atlas/>`__ with
  an Atlas cluster tier of M10 or higher.
```

Author: Updated.

> - A local deployment that is set up using the `Atlas CLI
>   <https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-deploy-local/>`__.

Reviewer: [suggestion] for minimalism

```suggestion
- A local deployment that uses the `Atlas CLI
  <https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-deploy-local/>`__.
```

Author: In this case, we want to point out that it has to be "set up" with Atlas CLI so I may go ahead and keep this in.

Additionally, your deployment must run MongoDB version 7.0 or later.

Steps
-----

.. procedure::
   :style: normal
   
   .. step:: Open the index creation dialog

>       From the :ref:`Indexes <indexes-tab>` tab, click the
>       :guilabel:`Create` button, then click :guilabel:`Search Index`.

Reviewer: ```suggestion
      From the :ref:`Indexes <indexes-tab>` tab, click the
      :guilabel:`Create` button and then click :guilabel:`Search Index`.
```

Author: Updated.

   .. step:: Specify a name for the index

   .. step:: Select the Atlas Search Index type

      Select :guilabel:`Vector Search`.

   .. step:: Provide the Atlas Vector Search Index configurations

>       You must initially provide the following default vector search index 
>       configurations. You can modify this later.

Reviewer: ```suggestion
      You must initially provide the following default vector search index 
      configurations. You can modify the configurations later.
```

Author: Updated.

      .. list-table::
         :header-rows: 1
         :widths: 20 10 70

         * - Field
           - Type
           - Description

         * - ``path``
           - string
           - The field name to index.
        
         * - ``numDimensions``
           - int 
           - The number of vector dimensions, which Atlas Search enforces at index- and 
             query-time. This value can't be greater than 2048.
        
         * - ``similarity``
           - string

>            - The vector similarity function used to search for top K-nearest neighbors.

Reviewer: ```suggestion
           - The vector similarity function used to search for the top K-nearest neighbors.
```

Author: Updated.

>              Select from the following:

Reviewer: ```suggestion
             Select from the following functions:
```

Author: Updated.

             .. list-table::
                :header-rows: 1
                :widths: 50 50

                * - Function
                  - Description

                * - ``euclidean``
                  - A function that measures the distance between ends of vectors. This allows 

>                     you to measure similarity based on varying dimensions. 

Reviewer: [suggestion] I suggest replacing "This allows" to "This function allows" for clarity on all the table items here

Author: Update all the instances of this.

                * - ``cosine``
                  - A function that measures similarity based on the angle between vectors. This 
                    allows you to measure similarity that isn't scaled by magnitude. You can't use 
                    zero magnitude vectors with cosine. To measure cosine similarity, we recommend 

>                     that you normalize your vectors and use dotProduct instead.

Reviewer: [suggestion/question] Do you think it would be useful to put the zero magnitude vector limitation in a note as opposed to in the main body of this list item? I'm wondering how often this use case might come up and whether users will notice the limitation if It is not highlighted.

Author: updated.

                * - ``dotProduct``

>                   -  A function that measures similar to cosine, but takes into account the 

Reviewer: ```suggestion
                  -  A function that measures similarly to cosine, but takes into account the 
```

Author: Updated.

                     magnitude of the vector. This allows you to efficiently measure similarity 
                     based on both angle and magnitude. To use dotProduct, you must normalize 
                     the vector to unit length at index- and query-time.

   .. step:: Click :guilabel:`Create Search Index`

Example
-------

> The following example definition uses the ``sample_mflix.embedded_movies`` collection 
> and indexes the ``plot_embedding`` field: 

Reviewer: [suggestion] I would suggest adding more context around what the example shows

```suggestion
The following example definition uses the ``sample_mflix.embedded_movies`` collection 
and indexes the ``plot_embedding`` field. This code block illustrates an example Atlas Vector Search configuration
```

Author: Updated to something similar.

.. code-block:: javascript

   {
      "fields": [ {
         "type": "vector",
         "path": "plot_embedding",
         "numDimensions": 1536,
         "similarity": "euclidean"
      } ]
   }

Results
-------

To view the status of your created index, go to the :guilabel:`Indexes`
tab and set the toggle at the top-right to :guilabel:`Search Indexes`.

The :guilabel:`Status` column indicates the status of the index. When

> the status is :guilabel:`Ready`, your index is ready to be used.

Reviewer: ```suggestion
the status is :guilabel:`Ready`, your index is ready to use.
```

Author: Updated.

.. _compass-create-vector-search:

===================================
Create an Atlas Vector Search Index
===================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

.. versionadded:: 1.42.1

> You can create Atlas Vector Search indexes using |compass|. These indexes 

Reviewer 2: Would it make sense to link to the general documentation on Atlas Vector Search indexes here?
https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/

Author: Yes!
